src/Orleans.Runtime/Streams/QueueBalancer/LeaseBasedQueueBalancer.cs

```diff
@@ -113,32 +109,33 @@ public LeaseBasedQueueBalancer(LeaseBasedQueueBalancerOptions options, IServiceP
             this.isStarting = true;
             this.loggerFactory = loggerFactory;
             this.options = options;
-            this.logger = loggerFactory.CreateLogger<LeaseBasedQueueBalancer>();
+            this.logger = loggerFactory.CreateLogger($"{typeof(LeaseBasedQueueBalancer).FullName}-{name}");
         }
 
         public static IStreamQueueBalancer Create(IServiceProvider services, string name, IDeploymentConfiguration deploymentConfiguration)
         {
-            var options = services.GetService<IOptionsSnapshot<LeaseBasedQueueBalancerOptions>>().Get(name);
-            return ActivatorUtilities.CreateInstance<LeaseBasedQueueBalancer>(services, options, deploymentConfiguration);
+            var options = services.GetRequiredService<IOptionsSnapshot<LeaseBasedQueueBalancerOptions>>().Get(name);
+            if (options == null)
```

Comment:
if there's no `LeaseBasedQueueBalancerOptions` with this name registered. would IOptionSnapshot return a default one or null? if return default one, then this null check isn't effective, right?

Reply:
yup.. I just moved this logic from Initialize.. didn't really look too hard at it.. bleh..
Will fix.